mlflow/models/evaluation/default_evaluator.py

```diff
@@ -369,6 +370,24 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _compute_df_mode_or_mean(df):
+    """
+    Compute mode (for integer or non-numeric columns or mean (for float columns) for the
+    input dataframe, return a dict, key is column name, value is the corresponding mode or
+    mean value.
+    """
+    # convert pandas dataframe columns to best possible dtypes.
+    # e.g. if one column has float type values, but they are actually all integers,
+    # it will be converted to integer column
+    df = df.convert_dtypes()
```

Comment:
Do we really need this conversion? Let's say we have a float feature and it only contains integers (e.g. `[1.0, 2.0, 5.0]`)) by coincidence. Should we really use mode for this feature?

Reply:
This case I think we should treat it as "integer" feature. The model might use this feature as categorical feature, but if we compute "mean", then we might generate a number which is not integer, then it might break the model prediction.

Reply:
I think most model infrastructure that does dynamic casting internally uses a cardinality check, right? Would it be better to handle the mean / mode validation for numeric types by using a similar approach?

Reply:
I think the one that you implemented for determining if a model is a classifier or a regressor for the evaluator API will work great!